torchrl/collectors/collectors.py

```diff
@@ -124,6 +158,46 @@ def _get_policy_and_device(
                     "env must be provided to _get_policy_and_device if policy is None"
                 )
             policy = RandomPolicy(self.env.action_spec)
+        elif isinstance(policy, nn.Module):
+            # TODO: revisit these checks when we have determined whether arbitrary
+            # callables should be supported as policies.
```

Comment:
More thought is going into whether arbitrary callables should be supported as policies. Once there is a clear answer on this question these checks should perhaps be revisited.

Reply:
yes I think that this is cool
if it's not a nn.Module we assume that it works out of the box